nextstrain/cli/command/login.py

```diff
@@ -0,0 +1,153 @@
+"""
+Logs in a Nextstrain.org groups user, storing their credentials locally.
+"""
+import os
+import ast
+import sys
+import jose
+import boto3
+import getpass
+import configparser
+from pathlib import Path
+from pycognito import Cognito
+from pycognito.aws_srp import AWSSRP
+from typing import Dict
+
+base = Path(__file__).resolve().parent.parent.parent.parent
+secrets_path = base / '.nextstrain/secrets'
+config = configparser.ConfigParser()
+
+def register_parser(subparser):
+    parser = subparser.add_parser("login", help = "Login as a Nextstrain.org groups user")
+
+    register_arguments(parser)
+
+    return parser
+
+
+def register_arguments(parser):
+    parser.add_argument(
+        "--check",
+        help    = "Check credentials by logging into nextstrain.org",
+        action  = 'store_true')
+
+    parser.add_argument(
+        "--delete",
+        help    = "Deletes the secrets file containing authentication credentials",
+        action  = 'store_true')
+
+    return parser
+
+
+def run(opts):
+    if opts.delete:
+        try:
+            os.remove(secrets_path)
+            print(f"Deleted the credentials file at {secrets_path}.")
+
+        except FileNotFoundError:
+            print("No credentials found to delete!")
+
+        sys.exit(0)
+
+    if opts.check:
+        print("Checking credentials by logging into nextstrain.org...")
+        try:
+            config.read(secrets_path)
+            username = config.get('login', 'username')
+            authn_result = ast.literal_eval(config.get('login', 'authenticationresult'))
+
+        # If the requested credentials are not found in a 'login' section in the
+        # specified secrets file, generate them, and save them locally
+        except (FileNotFoundError, configparser.NoSectionError, configparser.NoOptionError):
+            login()
+            sys.exit(0)
+
+        u = Cognito(os.environ['AWS_COGNITO_USER_POOL_ID'],
```

Comment:
The user pool id should be hardcoded in source, not come from the environment. Remember that the CLI runs on many people's computers, and no one will have or should have to have a specific environment variable set. (If necessary, we could support _overriding_ it from the environment, for things like testing/dev, but I don't think that's necessary here.)

Best to declare the hardcoded value as `COGNITO_USER_POOL_ID` at the top of the file.

Reply:
Gotcha, I wasn't sure if it was a secret or not, so I erred on the safer side.

Reply:
:+1: That's a good instinct! In our case, it's not a secret. (Other Cognito use cases may treat it as a semi-secret.)

For reference, you can also see what other Cognito values we treat as public or not for nextstrain.org authentication:

https://github.com/nextstrain/nextstrain.org/blob/497d62e30983042621436a13af6a6df21d48b166/authn.js#L27-L35

Reply:
Thanks for pointing out that file!